src/t_cose_sign1_sign.c

```diff
@@ -316,13 +327,12 @@ t_cose_sign1_encode_signature_aad_internal(struct t_cose_sign1_sign_ctx *me,
     struct q_useful_buf_c        tbs_hash;
     /* Pointer and length of the completed signature */
     struct q_useful_buf_c        signature;
-    /* Buffer for the actual signature */
-    Q_USEFUL_BUF_MAKE_STACK_UB(  buffer_for_signature, T_COSE_MAX_SIG_SIZE);
+    /* Pointer and length of the buffer for the signature */
+    struct q_useful_buf          buffer_for_signature;
```

Comment:
I may be missing something, but I can't find any initialization code for `buffer_for_signature` before the call to `QCBOREncode_OpenBytes` down below.

Reply:
`QCBOREncode_OpenBytes` does the initialization of it. It will make `buffer_for_signature` point to the space that is inside the CBOR buffer.

Reply:
Got it.